README.md

```diff
@@ -0,0 +1,743 @@
+- The request URL is on the same host as the current Backbone application.
+
+[bb-sync]: https://backbonejs.org/#Sync
+[csrf]: https://en.wikipedia.org/wiki/Cross-site_request_forgery#Cookie-to-header_token
+[django]: https://www.djangoproject.com/
+[jq-xhr]: https://api.jquery.com/jQuery.ajax/#jqXHR
+
+### Module `future-attribute`
+
+[view source](https://github.com/CentreForDigitalHumanities/backbone-util/tree/main/src/future-attribute.js)
+
+This module provides two ways to postpone a callback until a specific attribute on a model switches from unset to set. The callback is scheduled or invoked immediately if the attribute is already set at the moment of querying. This saves you from writing an `if`/`else` every time a callback depends on the presence of an attribute that might or might not be already set.
+
+```javascript
+import { Model, View } from 'backbone';
+import { when, whenever } from '@uu-cdh/backbone-util';
+
+var BookModel = Model.extend({urlRoot: '/book'});
+var bookInstance = new BookModel({id: 1});
+
+bookInstance.fetch();
+// This will take some time. In the meanwhile, we can already
+// schedule some code:
+
+when(bookInstance, 'title', function(book, title) {
+    // This callback will run once, when the book has a title.
+    alert('Fetched title for book "' + title + '"');
+});
+
+whenever(bookInstance, 'title', function(book, title) {
+    // This callback will run when the book has a title, and
+    // again every time the title changes.
+    $('head title').text(title);
+});
+```
+
+#### Function `when`
+
+**Named export** of `@uu-cdh/backbone-util/src/future-attribute.js`, **reexported by name** from the package index.
+
+**Parameters:**
+
+- `model`, the model instance on which you expect the attribute to appear.
+- `attribute`, the name of the attribute that you expect to be eventually set.
+- `handler`, the callback that should be invoked when the attribute is first set. It receives the same `(model, value, options)` arguments as a `model.on('change:attribute')` callback. `options` is an empty object if `attribute` was already set before `when` was called.
+- `context`, optional `this` binding for the `handler`. If unspecified, `model` is used.
+
+**Return value:** none
+
+**Side effect:** either schedules the `handler` for a later cycle of the event loop, if the `attribute` is already set, or registers `handler` as a one-time event handler for `'change:attribute'`, if the `attribute` is not yet set. If `context` is defined and it implements the [`Events`][bb-events] interface, `context.listenToOnce` is used to register the handler, otherwise `model.once`.
+
+It is important to realize that the `handler` *always runs async*, if it runs at all. In other words, it never runs before the call to `when` ends. This is one major difference from `whenever`, discussed next. The other major difference is that `handler` runs at most once, while `whenever` might cause it to run multiple times.
```

Comment:
To be honest, I feel a little bit unsure about this, but I cannot think of an alternative that is evidently more correct. The following reasons all seem to point to the current behavior:

- The callback of `when` runs only once. The recommended practice for callbacks is that they should run either always sync or always async.
- The callback of `whenever` has no limit to the number of calls. As such, if the attribute is already set and you defer the callback, an intervening `change` event might cause the callback to run "again" *before* the deferred callback gets an opportunity to run. This would lead to inconsistent ordering of callback invocations.
- `whenever` is basically a shorthand for `if (model.has(attribute)) callback(); model.on(`change:${attribute}`, callback);` (omitting some details). That first conditional invocation runs sync, too.